src/GT/Strava.hs

```diff
@@ -0,0 +1,97 @@
+{-# LANGUAGE RankNTypes      #-}
+{-# LANGUAGE RecordWildCards #-}
+
+-- | Import data from Strava.
+--
+-- This module uses the @Strive@ library.
+module GT.Strava where
+
+import GT.Prelude
+
+import           Control.Monad.Logger
+import           Data.Maybe
+import           Data.String
+import qualified Data.Text            as Text
+import           Database.Persist.Sql as DB
+import           Strive               hiding
+    ( map
+    )
+
+import qualified GT.DB.Schema.StravaActivity as DB
+import qualified GT.DB.Schema.StravaGear     as DB
+
+-- | Scaffolding. Ignore this.
+type Runner = forall x m. SqlPersistT m x -> m x
+
+-- | A draft implementatino of gear and activity import. This
+-- implementation does not currently handle Strava's rate limiting.
+runImport
+    :: (MonadLogger m, MonadIO m)
+    => Runner
+    -> Text -- ^ Strava authorization token
+    -> m ()
+runImport runDB token = do
+    logInfoN "Beginning import. . ."
+
+    client <- liftIO $ buildClient (Just token)
+    logDebugN "Created client."
+
+    loop client 1
+  where
+    loop client pageNumber = do
+        eresult <- liftIO $
+            getCurrentActivities client (with [set page pageNumber, set perPage 200])
+
+        case eresult of
+            Left (_resp, str) -> do
+                logErrorN $
+                    "Received Left from getCurrentActivities: " <> fromString str
+            Right [] ->
+                logInfoN "We're done!"
```

Comment:
Strava's documentation explicitly says that they'll return a response with `length results < pageSize`, for any reason at all really. The only way to be sure is if they return the empty list.